### PR TITLE
net:tc add dup() testcase

### PR DESCRIPTION
**Pull request by pillip8282**
Add dup testcases for socket desciptors.
It test unit test case at os/net/socket/net_dupsd* files

### CODE REVIEW

File: apps/examples/testcase/le_tc/network/tc_net_dup.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+#include "tc_internal.h"
+
+#define UTC_NET_LOGE printf
+
+/**
+   * @testcase		   :tc_net_dup_n
+   * @brief		       :
+   * @scenario		   :
+   * @apicovered	   :dup()
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/network/tc_net_dup.c]: fix allignment

File: apps/examples/testcase/le_tc/network/tc_net_dup.c [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	int fd = dup(sock);
+	TC_ASSERT_EQ("dup", fd, -1);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+   * @testcase		   :tc_net_dup_p
+   * @brief		       :
+   * @scenario		   :
+   * @apicovered	   :dup()
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/network/tc_net_dup.c]: same as above

File: apps/examples/testcase/le_tc/network/tc_net_dup.c [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	close(fd);
+	TC_ASSERT_GT("dup", fd, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+   * @testcase		   :tc_net_dup2_n
+   * @brief		       :
+   * @scenario		   :
+   * @apicovered	   :dup2()
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/network/tc_net_dup.c]: fix alignment in all comment headers also

File: apps/examples/testcase/le_tc/network/tc_net_dup.c [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+   * @precondition	   :socket()
+   * @postcondition	   :
+   */
+static void tc_net_dup2_p(void)
+{
+	int sock_old = socket(AF_INET, SOCK_STREAM, 0);
+	if (sock_old == -1) {
+		UTC_NET_LOGE("get old socket error (%d)\n", errno);
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/network/tc_net_dup.c]: Is it intentional to use `UTC_NET_LOGE` for `socket `API failure?  we can use `TC_ASSERT_XX` right?

File: apps/examples/testcase/le_tc/network/tc_net_dup.c (hunk 1 above)
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/tc_net_dup.c]: done

File: apps/examples/testcase/le_tc/network/tc_net_dup.c (hunk 2 above)
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/tc_net_dup.c]: done

File: apps/examples/testcase/le_tc/network/tc_net_dup.c (hunk 3 above)
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/tc_net_dup.c]: done

File: apps/examples/testcase/le_tc/network/tc_net_dup.c (hunk 4 above)
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/tc_net_dup.c]: yes, it's intentional because the error is not caused by the test case which I want to check

Comment (lokeshbv) [reply apps/examples/testcase/le_tc/network/tc_net_dup.c]: Ok

**Pull request merged**